src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -1199,29 +1199,28 @@ private async Task<CompilationCompletedEvent> ProcessCompilationEventsCoreAsync(
                     }
 
                     CompilationEvent e;
-                    try
+                    if (!CompilationEventQueue.TryDequeue(out e))
                     {
-                        if (!CompilationEventQueue.TryDequeue(out e))
+                        if (!prePopulatedEventQueue)
                         {
-                            if (!prePopulatedEventQueue)
-                            {
-                                e = await CompilationEventQueue.DequeueAsync(cancellationToken).ConfigureAwait(false);
-                            }
-                            else
+                            var optionalEvent = await CompilationEventQueue.TryDequeueAsync(cancellationToken).ConfigureAwait(false);
+                            if (!optionalEvent.HasValue)
                             {
-                                return completedEvent;
+                                // When the queue is completed with a pending TryDequeueAsync return, the
+                                // the Optional<T> will not have a value. This signals the queue has reached
+                                // completion and no more items will be added to it.
+
+                                // This failure is being tracked by https://github.com/dotnet/roslyn/issues/5962
```

Comment:
>This failure is being tracked by [](start = 35, length = 32)

This comment appears pretty out of date. Can this assertion be uncommented? At the very least we should have an open issue tracking this potential failure.

Reply:
I'll submit this as a follow-up